cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/TetheringAgentService.java

```diff
@@ -102,31 +103,30 @@
   private final TransactionRunner transactionRunner;
   private final Map<String, String> lastMessageIds;
   private final ProgramStateWriter programStateWriter;
-  private final MessageFetcher messageFetcher;
   private final ProgramRunRecordFetcher runRecordFetcher;
   private final RemoteAuthenticator remoteAuthenticator;
-  private final String programUpdateTopic;
-  private final int programUpdateFetchSize;
-  private String lastProgramUpdateMessageId;
+  private final RetryStrategy retryStrategy;
 
   @Inject
   TetheringAgentService(CConfiguration cConf, TransactionRunner transactionRunner, TetheringStore store,
                         ProgramStateWriter programStateWriter, MessagingService messagingService,
                         ProgramRunRecordFetcher programRunRecordFetcher,
+                        MetricsCollectionService metricsCollectionService,
                         @Named(REMOTE_TETHERING_AUTHENTICATOR) RemoteAuthenticator remoteAuthenticator) {
-    super(RetryStrategies.fromConfiguration(cConf, "tethering.agent."));
+    super("tethering.agent", cConf, cConf.get(Constants.AppFabric.PROGRAM_STATUS_RECORD_EVENT_TOPIC),
+          cConf.getInt(Constants.AppFabric.STATUS_EVENT_FETCH_SIZE),
+          cConf.getLong(Constants.AppFabric.STATUS_EVENT_POLL_DELAY_MILLIS), messagingService,
+          metricsCollectionService, transactionRunner);
+    this.retryStrategy = RetryStrategies.fromConfiguration(cConf, "tethering.agent.");
     this.connectionInterval = TimeUnit.SECONDS.toMillis(cConf.getLong(Constants.Tethering.CONNECTION_INTERVAL));
```

Comment:
don't think this needs to be done manually anymore, the superclass should take care of it.

Reply:
the superclass polls immediately, and since this is connecting to tethering server during each run, that seems too frequent
https://github.com/cdapio/cdap/blob/52005e2bb1bc14b5bac4bc011fc9e34c02cc61d7/cdap-tms/src/main/java/io/cdap/cdap/messaging/subscriber/AbstractMessagingPollingService.java#L184-L219